### DISCUSSION TITLE
USB Bluetooth 5.3 Adapter - Baseus BA07 on Void Linux

### DISCUSSION THREAD

**Discussion by MatanzaBR** [blueman-project/blueman#2313] [Q&A]
**Adaptador USB Bluetooth 5.3 - Baseus BA07 no Void Linux**

The Baseus BA07 works with a Windows 10 Pro 64 bit driver. In Windows, the sound works through headphones and through the speaker, files are sent and received from the cell phone.

Blueman shows the devices but there is no communication, it does not send or receive files from the cell phone.

Does not send audio to JBL headphones or JBL speakers.

Nothing about a Bluetooth device appears in pavucontrol.

Does not work Baseus BA07 on Linux (10d7:b012).

Any support for this Bluetooth device on Linux?

[https://www.baseusbr.com/produtos/adaptador-usb-bluetooth-5-3/](https://www.baseusbr.com/produtos/adaptador-usb-bluetooth-5-3/)

$ uname -r
6.6.22_1

$ uname -m
x86_64

Desktop environment: Xfce4

$ xbps-query -Rs blueman
[*] blueman-2.3.5_2 GTK+ Bluetooth Manager

$ xbps-query -Rs bluez
[*] bluez-5.70_2              Bluetooth tools and daemons

$ xbps-query -Rs pavucontrol
[*] pavucontrol-5.0_3      PulseAudio Volume Control

$ xbps-query -Rs Pulseaudio
[*] pulseaudio-16.1_2                   Featureful, general-purpose sound server

$ groups
lp  bluetooth  pulse-access

$ lsusb | grep adapter
Bus 004 Device 003: ID **10d7:b012 Actions general adapter**

$ lsusb -v
...

Bus 004 Device 003: ID 10d7:b012 Actions general adapter
Couldn't open device, some information will be missing
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass          224 Wireless
  bDeviceSubClass         1 Radio Frequency
  bDeviceProtocol         1 Bluetooth
  bMaxPacketSize0        64
  idVendor           0x10d7 Actions
  idProduct          0xb012 general adapter
  bcdDevice           88.91
  iManufacturer           1 Actions
  iProduct                2 general adapter
  iSerial                 3 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x00b1
    bNumInterfaces          2
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xe0
      Self Powered
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           2
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0000  1x 0 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0000  1x 0 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       1
      bNumEndpoints           2
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0009  1x 9 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0009  1x 9 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       2
      bNumEndpoints           2
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0011  1x 17 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0011  1x 17 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       3
      bNumEndpoints           2
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0019  1x 25 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0019  1x 25 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       4
      bNumEndpoints           2
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0021  1x 33 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0021  1x 33 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       5
      bNumEndpoints           2
      bInterfaceClass       224 Wireless
      bInterfaceSubClass      1 Radio Frequency
      bInterfaceProtocol      1 Bluetooth
      iInterface              0 
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x01  EP 1 OUT
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0031  1x 49 bytes
        bInterval               1
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            1
          Transfer Type            Isochronous
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0031  1x 49 bytes
        bInterval               1

**Comment by cschramm** [comment]
Doesn't seem (well) supported by Linux, yes, cp. https://bugzilla.kernel.org/show_bug.cgi?id=217870

**Comment by nukelet** [comment]
Just an FYI, I submitted a kernel patch last month to fix this issue: https://lore.kernel.org/all/20240227014328.1052386-1-nukelet64@gmail.com/

It has already landed in 6.9 and is queued for backporting to 6.8 and 6.6, so you can either wait a little for it to arrive in Void Linux or recompile the kernel yourself.

**Comment by diegok3** [comment]
If you have Debian or Ubuntu you can install xanmod kernel https://xanmod.org
For me (Debian 12.6) with this xankod kernel "Linux n 6.9.9-x64v4-xanmod1" Its working:

Follow xanmod install instructions but put last version:  linux-xanmod-x64v4 instead of linux-xanmod-x64v3

**Comment by MatanzaBR** [comment]
**Bluetooth 5.3 USB Adapter - Baseus BA07 working on Void Linux using kernel 6.6.39_1**


lsusb 

Bus 008 Device 007: ID **10d7:b012** Actions general adapter


cat /etc/os-release 
NAME="Void"
ID="void"
PRETTY_NAME="**Void Linux**"
HOME_URL="https://voidlinux.org/"
DOCUMENTATION_URL="https://docs.voidlinux.org/"
LOGO="void-logo"
ANSI_COLOR="0;38;2;71;128;97"

DISTRIB_ID="void"


**kernel:**
uname -r
**6.6.39_1**

uname -m
x86_64


xbps-query -l | grep blue
ii blueman-2.4.2_1                                         GTK+ Bluetooth Manager
ii bluez-5.76_1                                                Bluetooth tools and daemons
ii bluez-alsa-3.1.0_1                                       Bluetooth Audio ALSA Backend
ii libbluetooth-5.76_1                                    Library to use the Bluez Linux Bluetooth Stack


xbps-query -l | grep alsa
ii alsa-lib-1.2.11_1                                           Advanced Linux Sound Architecture (ALSA) library
ii alsa-plugins-pulseaudio-1.2.12_1             Advanced Linux Sound Architecture (ALSA) extra plugins - Pulseaudio output plugin
ii alsa-utils-1.2.11_1                                        Advanced Linux Sound Architecture (ALSA) utilities
ii bluez-alsa-3.1.0_1                                        Bluetooth Audio ALSA Backend
ii xfce4-alsa-plugin-0.3.0_1                           Simple ALSA volume control for xfce4-panel


xbps-query -l | grep pulseaudio
ii alsa-plugins-pulseaudio-1.2.12_1                  Advanced Linux Sound Architecture (ALSA) extra plugins - Pulseaudio output plugin
ii libpulseaudio-16.1_2                                        Featureful, general-purpose sound server - shared libraries
ii pulseaudio-16.1_2                                            Featureful, general-purpose sound server
ii pulseaudio-utils-16.1_2                                   Featureful, general-purpose sound server - utilities
ii xfce4-pulseaudio-plugin-0.4.8_1                   Pulseaudio volume control plugin for Xfce4 panel

**Comment by MatanzaBR** [comment]
**Product demo video on Void Linux**

Bluetooth 5.3 USB Adapter - Baseus BA07 working on Void Linux

https://www.youtube.com/watch?v=jJ8wBzLNp4o
